node/pkg/watchers/evm/watcher.go

```diff
@@ -745,6 +852,42 @@ func canRetryGetBlockTime(err error) bool {
 	return exists
 }
 
+// verifyAndPublish validates a MessagePublication to ensure that it's safe. If so, it broadcasts the message. This function
+// should be the only location where the watcher's msgC channel is written to.
+// Modifies the verificationState field of the message as a side-effect.
+// Even if an invalid Transfer is detected, the message will still be published. It is the responsibility of the calling code to handle
+// a status of Rejected.
+// Note that the result of verification is not returned by this function, but can be accessed directly via the reference to message.
+func (w *Watcher) verifyAndPublish(
+	// Must be non-nil and have verificationState equal to NotVerified.
+	msg *common.MessagePublication,
+	ctx context.Context,
+	// TODO: in practice it might be possible to read the txHash from the MessagePublication and so this argument might be redundant
+	txHash eth_common.Hash,
+	// This argument is only used when Transfer Verifier is enabled. If nil, transfer verifier will fetch the receipt.
+	// Otherwise, the receipt in the calling context can be passed here to save on RPC requests and parsing.
+	receipt *gethTypes.Receipt,
+) error {
+
+	if msg == nil {
+		return errors.New("verifyAndPublish: message publication cannot be nil")
+	}
+
+	if w.txVerifier != nil {
+		verifiedMsg, err := verify(ctx, msg, txHash, receipt, w.txVerifier)
+
+		if err != nil {
```

Comment:
I really think we should be pegging a metric or something if this fails. Otherwise how will anyone know it's happening?

Reply:
This is a good point and I'll resolve it in a follow-up PR. Now that we have this mechanism to mark messages as say `Anomalous`, the next task will be to figure out what we want to do with them. Based on that design, we can find good metrics for observing all of this.